com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/util/JarSignatureVerifier.java

```diff
@@ -142,21 +138,13 @@ private String getJarEntriesNames(List<JarEntry> jarEntries) {
 
     private void validateCertificateChain(List<X509Certificate> targetCertificates, List<X509Certificate> candidateCertificateChain,
                                           String certificateCN) {
-        boolean certificateCNMatches = false;
-        boolean targetCertificatesMatchEntryCertificate = false;
-        for (X509Certificate certificate : candidateCertificateChain) {
-            checkValidityOfCertificate(certificate);
-            if (!certificateCNMatches) {
-                certificateCNMatches = certificateCNMatches(certificate, certificateCN);
-            }
-            if (!targetCertificatesMatchEntryCertificate) {
-                targetCertificatesMatchEntryCertificate = doTargetCertificatesMatchEntryCertificate(targetCertificates, certificate);
-            }
-        }
-        if (!certificateCNMatches) {
+        candidateCertificateChain.forEach(this::checkValidityOfCertificate);
+
+        List<String> certificateCNs = getCertificatesNames(candidateCertificateChain);
+        if (certificateCN != null && !certificateCNs.contains(certificateCN)) {
             throw new SLException(Messages.WILL_LOOK_FOR_CERTIFICATE_CN, certificateCN);
         }
-        if (!targetCertificatesMatchEntryCertificate) {
+        if (!targetCertificates.containsAll(candidateCertificateChain)) {
```

Comment:
Old code checked for at least 1 match, this one checks for all matches. Add a test that breaks with this case + fix.